### ISSUE TITLE
from_env does not work when ENV_FOR_DYNACONF environment variable defined[bug]

### ISSUE THREAD

**Issue by talhajavaid32** [rochacbruno/dynaconf#276]
## The bug
`settings.from_env("<custom_env>")` does not switch to the __custom_env__ when the environment variable ENV_FOR_DYNACONF is defined with any environment.

### Directory Structure
.
├── .env
├── settings.yaml
└── test_dynaconf.py

### test_dynaconf.py contents

```python
from dynaconf import settings

print(settings.MODULE_OUTPUTS_BASE_DIR)
print(settings.OUTPUT_PATH)
print()
print("from env")
print()
other_settings = settings.from_env("OTHER")
print(other_settings.MODULE_OUTPUTS_BASE_DIR)
print(other_settings.OUTPUT_PATH)
```
### settings.yaml contents

```yaml
default:
    module_outputs_base_dir: temp/module_outputs/
    output_path: ./output/

pipeline_testing:
    module_outputs_base_dir: Cache/module_outputs_1/
    output_path: ../results/1/

other:
    module_outputs_base_dir: Cache/module_outputs_2/
    output_path: ../results/2/
```
### .env contents

```bash
ENV_FOR_DYNACONF=pipeline_testing
```

## Expected Behavior
Below is the expected output (which is attained only when there is no ENV_FOR_DYNACONF defined with the values before __from env__ statement being that from default)
```
Cache/module_outputs_1/
../results/1/

from env

Cache/module_outputs_2/
../results/2/
```

## Current Behavior
```
Cache/module_outputs_1/
../results/1/

from env

Cache/module_outputs_1/
../results/1/
```
## Environment
- OS: Linux/Ubuntu16.04
- Dynaconf Version 2.2.0

**Comment by rochacbruno** [comment]
this has been solved as part of #266 solution in #269 PR you must upgrade do Dynaconf 2.2.1 

can you please verify if upgrading fixes your use case?

**Comment by talhajavaid32** [comment]
Resolved on updating to 2.2.1. Thanks a lot
